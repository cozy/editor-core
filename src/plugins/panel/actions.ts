import {
  removeParentNodeOfType,
  findSelectedNodeOfType,
  removeSelectedNode,
  findParentNodeOfType,
} from 'prosemirror-utils';
import { NodeSelection } from 'prosemirror-state';
import { Node as PMNode } from 'prosemirror-model';
import { PanelType } from '@atlaskit/adf-schema';
import { getPanelTypeBackground } from '@atlaskit/editor-common';
import { Command } from '../../types';
import {
  AnalyticsEventPayload,
  ACTION,
  ACTION_SUBJECT,
  INPUT_METHOD,
  EVENT_TYPE,
  addAnalytics,
} from '../analytics';
import { findPanel } from './utils';
import { PanelOptions } from './pm-plugins/main';

export type DomAtPos = (pos: number) => { node: HTMLElement; offset: number };

export const removePanel = (): Command => (state, dispatch) => {
  const {
    schema: { nodes },
    tr,
  } = state;
  const payload: AnalyticsEventPayload = {
    action: ACTION.DELETED,
    actionSubject: ACTION_SUBJECT.PANEL,
    attributes: { inputMethod: INPUT_METHOD.TOOLBAR },
    eventType: EVENT_TYPE.TRACK,
  };

  let deleteTr = tr;
  if (findSelectedNodeOfType(nodes.panel)(tr.selection)) {
    deleteTr = removeSelectedNode(tr);
  } else if (findParentNodeOfType(nodes.panel)(tr.selection)) {
    deleteTr = removeParentNodeOfType(nodes.panel)(tr);
  }

  if (!deleteTr) {
    return false;
  }

  if (dispatch) {
    dispatch(addAnalytics(state, deleteTr, payload));
  }
  return true;
};

const getNewPanelData = (
  node: PMNode,
  newPanelType: PanelType,
  panelOptions: PanelOptions = {},
): { panelIcon: string; panelColor: string; panelType: PanelType } => {
  let {
    panelIcon: previousIcon,
    panelColor: previousColor,
    panelType: previousType,
  } = node.attrs;
  const { emoji, color } = panelOptions;

  let panelIcon = previousIcon;
  let panelType = newPanelType;

  let panelColor = getPanelTypeBackground(
    (newPanelType !== PanelType.CUSTOM
      ? newPanelType
      : previousType) as Exclude<PanelType, PanelType.CUSTOM>,
  );

  if (color || previousColor) {
    panelColor = color || previousColor;
  }

  if (emoji) {
    panelIcon = emoji;
  }

  return {
    panelIcon,
    panelColor,
    panelType: panelType,
  };
};

export const changePanelType = (
  panelType: PanelType,
  panelOptions: PanelOptions = {},
  UNSAFE_allowCustomPanel: boolean = false,
): Command => (state, dispatch) => {
  const {
    schema: { nodes },
    tr,
  } = state;

  const panelNode = findPanel(state);
  if (panelNode === undefined) {
    return false;
  }

  let newType = panelType;
  let previousType = panelNode.node.attrs.panelType;

  if (UNSAFE_allowCustomPanel) {
    const { panelType: newPanelType } = getNewPanelData(
      panelNode.node,
      panelType,
      panelOptions,
    );
    newType = newPanelType;
    previousType = panelType;
  }

  const payload: AnalyticsEventPayload = {
    action: ACTION.CHANGED_TYPE,
    actionSubject: ACTION_SUBJECT.PANEL,
    attributes: { newType, previousType },
    eventType: EVENT_TYPE.TRACK,
  };

  let newTr;
  if (UNSAFE_allowCustomPanel) {
    const { panelIcon, panelColor, panelType: newPanelType } = getNewPanelData(
      panelNode.node,
      panelType,
      panelOptions,
    );

    newTr = tr.setNodeMarkup(panelNode.pos, nodes.panel, {
      panelType: newPanelType,
      panelIcon,
      panelColor,
    });
  } else {
    newTr = tr.setNodeMarkup(panelNode.pos, nodes.panel, { panelType });
  }

  // Select the panel if it was previously selected
  const newTrWithSelection =
    state.selection instanceof NodeSelection &&
    state.selection.node.type.name === 'panel'
      ? newTr.setSelection(new NodeSelection(tr.doc.resolve(panelNode.pos)))
      : newTr;

  const changePanelTypeTr = addAnalytics(state, newTrWithSelection, payload);

  changePanelTypeTr.setMeta('scrollIntoView', false);

  if (dispatch) {
    dispatch(changePanelTypeTr);
  }
  return true;
};
