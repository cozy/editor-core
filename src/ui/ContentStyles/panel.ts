import { css } from '@emotion/react';

import {
  panelSharedStyles,
  PanelSharedCssClassName,
} from '@atlaskit/editor-common/panel';
import {
  SelectionStyle,
  getSelectionStyles,
  akEditorDeleteBackground,
  akEditorDeleteBackgroundWithOpacity,
  akEditorDeleteBorder,
  akEditorSelectedBorderSize,
  akEditorDeleteIconColor,
  akEditorSelectedNodeClassName,
} from '@atlaskit/editor-shared-styles';
import type { ThemeProps } from '@atlaskit/theme/types';
import { token } from '@atlaskit/tokens';

export const panelStyles = (props: ThemeProps) => css`
  .ProseMirror {
    .${PanelSharedCssClassName.prefix} {
      cursor: pointer;

      /* Danger when top level node */
      &.danger {
        box-shadow: 0 0 0 ${akEditorSelectedBorderSize}px
          ${akEditorDeleteBorder};
        background-color: ${token(
          'color.background.danger',
          akEditorDeleteBackground,
        )} !important;

        .${PanelSharedCssClassName.icon} {
          color: ${token(
            'color.icon.danger',
            akEditorDeleteIconColor,
          )} !important;
        }
      }
    }

    .${PanelSharedCssClassName.content} {
      cursor: text;
    }

    /* Danger when nested node */
    .danger .${PanelSharedCssClassName.prefix} {
      &[data-panel-type] {
        background-color: ${token(
          'color.blanket.danger',
          akEditorDeleteBackgroundWithOpacity,
        )};

        .${PanelSharedCssClassName.icon} {
          color: ${token('color.icon.danger', akEditorDeleteIconColor)};
        }
      }
    }

    ${panelSharedStyles(props)};
  }

  .${PanelSharedCssClassName.prefix}.${akEditorSelectedNodeClassName}:not(.danger) {
    ${getSelectionStyles([SelectionStyle.BoxShadow, SelectionStyle.Blanket])}
  }
`;
