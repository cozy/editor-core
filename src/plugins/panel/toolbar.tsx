import {
  FloatingToolbarColorPicker,
  FloatingToolbarConfig,
  FloatingToolbarItem,
} from './../floating-toolbar/types';
import { PanelPluginOptions } from './types';
import { defineMessages, InjectedIntl } from 'react-intl';
import SuccessIcon from '@atlaskit/icon/glyph/editor/success';
import InfoIcon from '@atlaskit/icon/glyph/editor/info';
import NoteIcon from '@atlaskit/icon/glyph/editor/note';
import RemoveIcon from '@atlaskit/icon/glyph/editor/remove';
import WarningIcon from '@atlaskit/icon/glyph/editor/warning';
import ErrorIcon from '@atlaskit/icon/glyph/editor/error';
import { PanelType } from '@atlaskit/adf-schema';

import commonMessages from '../../messages';
import { removePanel, changePanelType } from './actions';
import { hoverDecoration } from '../base/pm-plugins/decoration';
import { EditorState } from 'prosemirror-state';
import { NodeType } from 'prosemirror-model';

import { Command } from '../../types';
import { panelBackgroundPalette } from '../../ui/ColorPalette/Palettes/panelBackgroundPalette';
import {
  ProviderFactory,
  getPanelTypeBackground,
} from '@atlaskit/editor-common';
import { FormattedMessage } from 'react-intl';
import { findPanel } from './utils';
import { EditorView } from 'prosemirror-view';
import { findDomRefAtPos } from 'prosemirror-utils';
import {
  DEFAULT_BORDER_COLOR,
  PaletteColor,
} from '../../ui/ColorPalette/Palettes';

export const messages = defineMessages({
  info: {
    id: 'fabric.editor.info',
    defaultMessage: 'Info',
    description:
      'Panels provide a way to highlight text. The info panel has a blue background.',
  },
  note: {
    id: 'fabric.editor.note',
    defaultMessage: 'Note',
    description:
      'Panels provide a way to highlight text. The note panel has a purple background.',
  },
  success: {
    id: 'fabric.editor.success',
    defaultMessage: 'Success',
    description:
      'Panels provide a way to highlight text. The success panel has a green background.',
  },
  warning: {
    id: 'fabric.editor.warning',
    defaultMessage: 'Warning',
    description:
      'Panels provide a way to highlight text. The warning panel has a yellow background.',
  },
  error: {
    id: 'fabric.editor.error',
    defaultMessage: 'Error',
    description:
      'Panels provide a way to highlight text. The error panel has a red background.',
  },
  emoji: {
    id: 'fabric.editor.panel.emoji',
    defaultMessage: 'Add icon',
    description: 'Select the panel icon',
  },
  backgroundColor: {
    id: 'fabric.editor.panel.backgroundColor',
    defaultMessage: 'Background color',
    description: 'Select the panel background color.',
  },
});

export const getToolbarItems = (
  formatMessage: (
    messageDescriptor: FormattedMessage.MessageDescriptor,
  ) => string,
  panelNodeType: NodeType,
  isCustomPanelEnabled: boolean,
  providerFactory: ProviderFactory,
  activePanelType?: string,
  activePanelColor?: string,
  activePanelIcon?: string,
): FloatingToolbarItem<Command>[] => {
  const items: FloatingToolbarItem<Command>[] = [
    {
      id: 'editor.panel.info',
      type: 'button',
      icon: InfoIcon,
      onClick: changePanelType(PanelType.INFO),
      selected: activePanelType === PanelType.INFO,
      title: formatMessage(messages.info),
    },
    {
      id: 'editor.panel.note',
      type: 'button',
      icon: NoteIcon,
      onClick: changePanelType(PanelType.NOTE),
      selected: activePanelType === PanelType.NOTE,
      title: formatMessage(messages.note),
    },
    {
      id: 'editor.panel.success',
      type: 'button',
      icon: SuccessIcon,
      onClick: changePanelType(PanelType.SUCCESS),
      selected: activePanelType === PanelType.SUCCESS,
      title: formatMessage(messages.success),
    },
    {
      id: 'editor.panel.warning',
      type: 'button',
      icon: WarningIcon,
      onClick: changePanelType(PanelType.WARNING),
      selected: activePanelType === PanelType.WARNING,
      title: formatMessage(messages.warning),
    },
    {
      id: 'editor.panel.error',
      type: 'button',
      icon: ErrorIcon,
      onClick: changePanelType(PanelType.ERROR),
      selected: activePanelType === PanelType.ERROR,
      title: formatMessage(messages.error),
    },
  ];

  if (isCustomPanelEnabled) {
    const changeColor = (color: string): Command => (state, dispatch) => {
      changePanelType(PanelType.CUSTOM, { color }, true)(state, dispatch);
      return false;
    };

    const changeEmoji = (emoji: string): Command => (state, dispatch) => {
      changePanelType(
        PanelType.CUSTOM,
        { emoji: emoji },
        true,
      )(state, dispatch);
      return false;
    };

    const panelColor =
      activePanelType === PanelType.CUSTOM
        ? activePanelColor || getPanelTypeBackground(PanelType.INFO)
        : getPanelTypeBackground(
            activePanelType as Exclude<PanelType, PanelType.CUSTOM>,
          );

    const defaultPalette =
      panelBackgroundPalette.find((item) => item.value === panelColor) ||
      ({
        label: 'Custom',
        value: panelColor,
        border: DEFAULT_BORDER_COLOR,
      } as PaletteColor);

    const colorPicker: FloatingToolbarColorPicker<Command> = {
      id: 'editor.panel.colorPicker',
      title: formatMessage(messages.backgroundColor),
      type: 'select',
      selectType: 'color',
      defaultValue: defaultPalette,
      options: panelBackgroundPalette,
      onChange: (option) => changeColor(option.value),
    };

    items.push(
      {
        type: 'separator',
      },
      colorPicker,
      {
        id: 'editor.panel.emojiPicker',
        title: formatMessage(messages.emoji),
        type: 'emoji-picker',
        selected: activePanelType === PanelType.CUSTOM,
        onChange: (emoji) => changeEmoji(emoji),
      },
    );
  }

  items.push(
    {
      type: 'separator',
    },
    {
      id: 'editor.panel.delete',
      type: 'button',
      appearance: 'danger',
      icon: RemoveIcon,
      onClick: removePanel(),
      onMouseEnter: hoverDecoration(panelNodeType, true),
      onMouseLeave: hoverDecoration(panelNodeType, false),
      title: formatMessage(commonMessages.remove),
    },
  );

  return items;
};

export const getToolbarConfig = (
  state: EditorState,
  intl: InjectedIntl,
  options: PanelPluginOptions = {},
  providerFactory: ProviderFactory,
): FloatingToolbarConfig | undefined => {
  const { formatMessage } = intl;
  const panelObject = findPanel(state);
  if (panelObject) {
    const nodeType = state.schema.nodes.panel;
    const { panelType, panelColor, panelIcon } = panelObject.node.attrs;

    // force toolbar to be turned on
    const items = getToolbarItems(
      formatMessage,
      nodeType,
      options.UNSAFE_allowCustomPanel || false,
      providerFactory,
      panelType,
      options.UNSAFE_allowCustomPanel ? panelColor : undefined,
      options.UNSAFE_allowCustomPanel ? panelIcon : undefined,
    );

    const getDomRef = (editorView: EditorView) => {
      const domAtPos = editorView.domAtPos.bind(editorView);
      const element = findDomRefAtPos(
        panelObject.pos,
        domAtPos,
      ) as HTMLDivElement;
      return element;
    };

    return {
      title: 'Panel floating controls',
      getDomRef,
      nodeType,
      items,
    };
  }
  return;
};
