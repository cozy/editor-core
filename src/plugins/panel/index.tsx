import React from 'react';
import { panel, customPanel, PanelType } from '@atlaskit/adf-schema';
import { QuickInsertActionInsert } from '@atlaskit/editor-common/provider-factory';
import { EditorState } from 'prosemirror-state';
import { EditorPlugin } from '../../types';
import { createPlugin } from './pm-plugins/main';
import { getToolbarConfig } from './toolbar';
import keymap from './pm-plugins/keymaps';
import {
  addAnalytics,
  ACTION,
  ACTION_SUBJECT,
  ACTION_SUBJECT_ID,
  INPUT_METHOD,
  EVENT_TYPE,
} from '../analytics';
import {
  IconPanel,
  IconPanelNote,
  IconPanelSuccess,
  IconPanelWarning,
  IconPanelError,
} from '../quick-insert/assets';
import { messages } from '../block-type/messages';
import { PanelPluginOptions } from './types';

const insertPanelTypeWithAnalytics = (
  panelType: PanelType,
  state: EditorState,
  insert: QuickInsertActionInsert,
) => {
  const tr = insert(insertPanelType(panelType, state));
  if (tr) {
    addAnalytics(state, tr, {
      action: ACTION.INSERTED,
      actionSubject: ACTION_SUBJECT.DOCUMENT,
      actionSubjectId: ACTION_SUBJECT_ID.PANEL,
      attributes: {
        inputMethod: INPUT_METHOD.QUICK_INSERT,
        panelType,
      },
      eventType: EVENT_TYPE.TRACK,
    });
  }
  return tr;
};

const insertPanelType = (panelType: PanelType, state: EditorState) =>
  state.schema.nodes.panel.createChecked(
    { panelType },
    state.schema.nodes.paragraph.createChecked(),
  );

const panelPlugin = (options: PanelPluginOptions = {}): EditorPlugin => ({
  name: 'panel',

  nodes() {
    //TODO: ED-10445 remove this check after emoji panels moved to full schema
    const panelNode = options.UNSAFE_allowCustomPanel ? customPanel : panel;
    return [{ name: 'panel', node: panelNode }];
  },

  pmPlugins() {
    return [
      {
        name: 'panel',
        plugin: ({ providerFactory, dispatch }) =>
          createPlugin(dispatch, providerFactory, options),
      },
      {
        name: 'panelKeyMap',
        plugin: () => keymap(),
      },
    ];
  },

  pluginsOptions: {
    quickInsert: ({ formatMessage }) => [
      {
        id: 'infopanel',
        title: formatMessage(messages.infoPanel),
        keywords: ['panel'],
        description: formatMessage(messages.infoPanelDescription),
        priority: 800,
        icon: () => <IconPanel label={formatMessage(messages.infoPanel)} />,
        action(insert, state) {
          return insertPanelTypeWithAnalytics(PanelType.INFO, state, insert);
        },
      },
      {
        id: 'notepanel',
        title: formatMessage(messages.notePanel),
        description: formatMessage(messages.notePanelDescription),
        priority: 1000,
        icon: () => <IconPanelNote label={formatMessage(messages.notePanel)} />,
        action(insert, state) {
          return insertPanelTypeWithAnalytics(PanelType.NOTE, state, insert);
        },
      },
      {
        id: 'successpanel',
        title: formatMessage(messages.successPanel),
        description: formatMessage(messages.successPanelDescription),
        keywords: ['tip'],
        priority: 1000,
        icon: () => (
          <IconPanelSuccess label={formatMessage(messages.successPanel)} />
        ),
        action(insert, state) {
          return insertPanelTypeWithAnalytics(PanelType.SUCCESS, state, insert);
        },
      },
      {
        id: 'warningpanel',
        title: formatMessage(messages.warningPanel),
        description: formatMessage(messages.warningPanelDescription),
        priority: 1000,
        icon: () => (
          <IconPanelWarning label={formatMessage(messages.warningPanel)} />
        ),
        action(insert, state) {
          return insertPanelTypeWithAnalytics(PanelType.WARNING, state, insert);
        },
      },
      {
        id: 'errorpanel',
        title: formatMessage(messages.errorPanel),
        description: formatMessage(messages.errorPanelDescription),
        priority: 1000,
        icon: () => (
          <IconPanelError label={formatMessage(messages.errorPanel)} />
        ),
        action(insert, state) {
          return insertPanelTypeWithAnalytics(PanelType.ERROR, state, insert);
        },
      },
    ],
    floatingToolbar: (state, intl, providerFactory) =>
      getToolbarConfig(state, intl, options, providerFactory),
  },
});

export default panelPlugin;
