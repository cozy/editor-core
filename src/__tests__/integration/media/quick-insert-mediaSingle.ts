import Page from '@atlaskit/webdriver-runner/wd-wrapper';
import { BrowserTestCase } from '@atlaskit/webdriver-runner/runner';
import {
  editable,
  getDocFromElement,
  insertMediaFromMediaPicker,
  quickInsert,
  fullpage,
} from '../_helpers';
import {
  mountEditor,
  goToEditorTestingExample,
} from '../../__helpers/testing-example-helpers';
import { selectors } from '../panel/_utils';

[
  {
    skipBrowsers: ['edge', 'safari'],
    message: 'Inserts media single before paragraph',
    setup: async (page: Page) => {
      await page.type(editable, 'some text');
      await page.keys(Array(9).fill('ArrowLeft'));
    },
  },
  {
    skipBrowsers: ['edge', 'safari'],
    message: 'Inserts media single after paragraph',
    setup: async (page: Page) => {
      await page.type(editable, 'some text ');
    },
  },
  {
    skipBrowsers: ['edge', 'safari'],
    message: 'Inserts media single before paragraph nested in a panel',
    setup: async (page: Page) => {
      await quickInsert(page, 'Info panel');
      await page.waitForSelector(selectors.PANEL_EDITOR_CONTAINER);
      await page.type(editable, 'some text');
      await page.keys(Array(9).fill('ArrowLeft'));
    },
  },
  {
    skipBrowsers: ['edge', 'safari'],
    message: 'Inserts a media single after paragraph nested in a panel',
    setup: async (page: Page) => {
      await quickInsert(page, 'Info panel');
      await page.waitForSelector(selectors.PANEL_EDITOR_CONTAINER);
      await page.type(editable, 'some text ');
    },
  },
].forEach(({ message, setup, skipBrowsers }) => {
  BrowserTestCase(
    `quick-insert-mediaSingle.ts: ${message}`,
    { skip: skipBrowsers as any },
    async (
      client: Parameters<typeof goToEditorTestingExample>[0],
      testName: string,
    ) => {
      const page = await goToEditorTestingExample(client);
      await mountEditor(page, {
        appearance: fullpage.appearance,
        media: {
          allowMediaSingle: true,
          allowMediaGroup: true,
        },
        allowPanel: true,
        allowNewInsertionBehaviour: true,
      });

      await page.click(editable);
      await setup(page);

      await quickInsert(page, 'Files & images');
      await insertMediaFromMediaPicker(page);

      const doc = await page.$eval(editable, getDocFromElement);
      expect(doc).toMatchCustomDocSnapshot(testName);
    },
  );
});
