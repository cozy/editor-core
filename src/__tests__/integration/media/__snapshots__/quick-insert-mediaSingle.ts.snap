// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`quick-insert-mediaSingle.ts: Inserts a media single after paragraph nested in a panel 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "panelType": "info",
      },
      "content": Array [
        Object {
          "content": Array [
            Object {
              "text": "some text ",
              "type": "text",
            },
          ],
          "type": "paragraph",
        },
      ],
      "type": "panel",
    },
    Object {
      "attrs": Object {
        "layout": "center",
        "width": null,
      },
      "content": Array [
        Object {
          "attrs": Object {
            "__contextId": "DUMMY-OBJECT-ID",
            "__displayType": null,
            "__external": false,
            "__fileMimeType": "image/svg+xml",
            "__fileName": "example.png",
            "__fileSize": 235,
            "alt": null,
            "collection": "MediaServicesSample",
            "height": 4096,
            "id": "1234-5678-abcd-efgh",
            "occurrenceKey": null,
            "type": "file",
            "url": null,
            "width": 4096,
          },
          "type": "media",
        },
      ],
      "type": "mediaSingle",
    },
  ],
  "type": "doc",
}
`;

exports[`quick-insert-mediaSingle.ts: Inserts media single after paragraph 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "text": "some text ",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
    Object {
      "attrs": Object {
        "layout": "center",
        "width": null,
      },
      "content": Array [
        Object {
          "attrs": Object {
            "__contextId": "DUMMY-OBJECT-ID",
            "__displayType": null,
            "__external": false,
            "__fileMimeType": "image/svg+xml",
            "__fileName": "example.png",
            "__fileSize": 235,
            "alt": null,
            "collection": "MediaServicesSample",
            "height": 4096,
            "id": "1234-5678-abcd-efgh",
            "occurrenceKey": null,
            "type": "file",
            "url": null,
            "width": 4096,
          },
          "type": "media",
        },
      ],
      "type": "mediaSingle",
    },
  ],
  "type": "doc",
}
`;

exports[`quick-insert-mediaSingle.ts: Inserts media single before paragraph 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "layout": "center",
        "width": null,
      },
      "content": Array [
        Object {
          "attrs": Object {
            "__contextId": "DUMMY-OBJECT-ID",
            "__displayType": null,
            "__external": false,
            "__fileMimeType": "image/svg+xml",
            "__fileName": "example.png",
            "__fileSize": 235,
            "alt": null,
            "collection": "MediaServicesSample",
            "height": 4096,
            "id": "1234-5678-abcd-efgh",
            "occurrenceKey": null,
            "type": "file",
            "url": null,
            "width": 4096,
          },
          "type": "media",
        },
      ],
      "type": "mediaSingle",
    },
    Object {
      "content": Array [
        Object {
          "text": "some text",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`quick-insert-mediaSingle.ts: Inserts media single before paragraph nested in a panel 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "layout": "center",
        "width": null,
      },
      "content": Array [
        Object {
          "attrs": Object {
            "__contextId": "DUMMY-OBJECT-ID",
            "__displayType": null,
            "__external": false,
            "__fileMimeType": "image/svg+xml",
            "__fileName": "example.png",
            "__fileSize": 235,
            "alt": null,
            "collection": "MediaServicesSample",
            "height": 4096,
            "id": "1234-5678-abcd-efgh",
            "occurrenceKey": null,
            "type": "file",
            "url": null,
            "width": 4096,
          },
          "type": "media",
        },
      ],
      "type": "mediaSingle",
    },
    Object {
      "attrs": Object {
        "panelType": "info",
      },
      "content": Array [
        Object {
          "content": Array [
            Object {
              "text": "some text",
              "type": "text",
            },
          ],
          "type": "paragraph",
        },
      ],
      "type": "panel",
    },
  ],
  "type": "doc",
}
`;
